src/gt4py/next/iterator/ir.py

```diff
@@ -38,6 +39,22 @@ def __hash__(self) -> int:
 
 class Sym(Node):  # helper
     id: Coerced[SymbolName]  # noqa: A003
+    # TODO(tehrengruber): Revisit. Using strings is a workaround to avoid coupling with the
+    #   type inference.
+    kind: Optional[typing.Literal["Iterator", "Value"]] = None
```

Comment:
FYI, `None` can also be used as literal value.
```suggestion
    kind: typing.Literal["Iterator", "Value", None] = None
```